server/sonar-server/src/main/java/org/sonar/server/component/db/ComponentDao.java

```diff
@@ -182,18 +181,36 @@ protected void doDeleteByKey(DbSession session, String key) {
   public List<ComponentDto> selectProvisionedProjects(DbSession session, SearchOptions searchOptions, @Nullable String query) {
     Map<String, String> parameters = new HashMap<>();
     parameters.put("qualifier", Qualifiers.PROJECT);
```

Comment:
:new: ![MINOR](https://raw.githubusercontent.com/henryju/image-hosting/master/minor.png)Define a constant instead of duplicating this literal "qualifier" 4 times.[![...](https://raw.githubusercontent.com/henryju/image-hosting/master/more.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1192)